text/0911-const-fn.md

```diff
@@ -172,6 +172,18 @@ const fn arithmetic_ops<T: Int>() -> [fn(T, T) -> T; 4] {
 }
 ```
 
+`const` functions can also be unsafe, allowing construction of types that require
+invariants to be maintained (e.g. `std::ptr::Unique` requires a non-null pointer)
+```rust
+struct OptionalInt(u32);
+impl OptionalInt {
+    /// Value must be non-zero
+    unsafe const fn new(val: u32) -> OptionalInt {
```

Comment:
“unsafe constant function” sounds more “right” to me than “constant unsafe function”, though your point is also valid.

Either way, I think it should be possible to put these in any order. I’ve already had some trouble remembering which goes first in `pub extern fn x` and now I’ll have to think what order `pub`, `const` and `unsafe` go in `pub unsafe const fn x`. `pub` is pretty easy since it “always goes first”, but there’s no such obvious rule for `unsafe` and `const`.